File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        obj = defaultdict(int)
        for n in nums:
            obj[n] += 1
        lst = list(obj.items())
        lst.sort(key=lambda x: x[1], reverse=True)
        res = []
        i = 0
        while i < k:
            res.append(lst[i][0])
            i += 1
        
        return res

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
    const map = new Map()

    for (const n of nums) {
        const val = (map.get(n) || 0) + 1
        map.set(n, val)
    }

    const res = []

    const vals = Array.from(map.entries());
    vals.sort((a, b) => b[1] - a[1]);
    let i = 0;
    while (i < k) {
        res.push(vals[i][0])
        i++;
    }
    return res
};